### PR TITLE
RPRBLND-1479: MaterialX network

### PR DESCRIPTION
**Pull request by bnagirniak**
### PURPOSE
Add support to create MaterialX network and nodes with generated node types. 

### EFFECT OF CHANGE
Added new MaterialX node tree. Available to create Standard Surface shader. 

### TECHNICAL STEPS
1. Added copying MaterialX libs and supported files in create_hdusd_libs.py script.
2. Added MaterialX nodetree as MxNodeTree class.
3. Implemented MaterialX node as MxNode from mx.NodeDef structure.
4. Added creation of Standard Surface shader from standard_surface.mtlx.
5. Made some code cleanup.

### NOTES FOR REVIEWERS
This is an intermediate PR, new MaterialX node tree network isn't functional. Other nodes and its functionality, material assigning etc propose to create in next tasks.

### CODE REVIEW

File: src/hdusd/mx_nodes/nodes/__init__.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from nodeitems_utils import NodeCategory, NodeItem
+
+from ...utils import HDUSD_LIBS_DIR
+from .. import log
+from .base_node import create_node_types, MxNodeSocket
+
+
+node_types = create_node_types([
+    HDUSD_LIBS_DIR / "materialx/libraries/bxdf/standard_surface.mtlx",
+    # HDUSD_LIBS_DIR / "materialx/libraries/pbrlib/pbrlib_defs.mtlx",
```
Comment (bsavery) [inline src/hdusd/mx_nodes/nodes/__init__.py:26]: Why are these ones skipped?

File: src/hdusd/mx_nodes/nodes/base_node.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    """Base node from which all MaterialX nodes will be made"""
+    bl_compatibility = {'HdUSD'}
+    bl_icon = 'MATERIAL'
+    bl_label = 'mx'
+
+    # holds the materialx nodedef object
+    mx_nodedef: mx.NodeDef
+
+    def init(self, context):
+        ''' generates inputs and outputs from ones specified '''
```
Comment (bsavery) [inline src/hdusd/mx_nodes/nodes/base_node.py:60]: ''' generates inputs and outputs from ones specified in the mx_nodedef'''

File: src/hdusd/mx_nodes/nodes/__init__.py (hunk 1 above)
Comment (bsavery) [reply src/hdusd/mx_nodes/nodes/__init__.py:26]: Maybe I did this.  https://github.com/materialx/MaterialX/blob/master/libraries/pbrlib/pbrlib_defs.mtlx but I think they're useful

Comment (bnagirniak) [reply src/hdusd/mx_nodes/nodes/__init__.py:26]: There are lot of nodes there and we need to think how to group them. This I plan to do in next task.
Also pbrlib_defs.mtlx has several interpretations of math nodes, for example 3 add nodes, 6 multiply nodes etc. Therefore I decided not to include it in this PR.
Also there are lot of node definitions in stdlib_defs.mtlx. This also will be included in next task.

**Comment by bnagirniak** [comment]
BTW, 'libs' folder is already updated with MaterialX files on CIS.

**Comment by bnagirniak** [comment]
Forgot to mention, I used MaterialX binaries from https://github.com/materialx/MaterialX/releases/download/v1.37.3/MaterialX_Windows_VS2017_x64_Python37.zip
Script create_hdusd_libs.py was adjusted to copy data from materialx libs dir.

**Review by ardraeiss — APPROVED** [review]
Code is ok

**Comment by bnagirniak** [comment]
Implementation was continued in #31 therefore closing this PR.

**Pull request closed without merging**